src/python/WMCore/MicroService/MSUnmerged/MSUnmergedRSE.py

```diff
@@ -43,6 +49,108 @@ def __init__(self, rseName, **kwargs):
                       "deletedFail": []},
             "dirs": {"allUnmerged": set(),
                      "toDelete": set(),
-                     "protected": set(),
-                     "empty": []}}
+                     "protected": set()}
+        }
         self.update(myDoc)
+        self.mongoFilter = {'name': self['name']}
+
+    def buildMongoProjection(self, dumpRSE=False):
+        """
+        Returns the correct mongoProjection based on the `dumpRSE` flag passed
+        :param dumpRSE: Flag to decide whether the whole RSE object is to be
+                        included in the projection ergo saved into the database.
+        :return:        A dictionary type MongoDB Projection
+        """
+        if dumpRSE:
+            mongoProjection = {
+                "_id": False,
+                "name": True,
+                "pfnPrefix": True,
+                "isClean" : True,
+                "timestamps": True,
+                "counters": True,
+                "files": True,
+                "dirs": True}
+        else:
+            mongoProjection = {
+                "_id": False,
+                "name": True,
+                "pfnPrefix": True,
+                "isClean" : True,
+                "timestamps": True,
+                "counters": True,
+                "files": False,
+                "dirs": True}
+        return mongoProjection
+
+    def readRSEFromMongoDB(self, collection):
+        """
+        A method to read the RSE object from Database and update it's fields.
+        :param collection: The MonogoDB collection to read from
+        :return:           True if read and update were both successful, False otherwise.
+        """
+        # NOTE: The fields to be manipulated are only those which are compatible
+        #       with MongoDB (i.e. here we avoid any field holding a strictly
+        #       pythonic objects, like generators etc.)
+        mongoRecord = collection.find_one(self.mongoFilter)
+
+        # update the list fields read from MongoDB back to strictly pythonic `set`s
+        if mongoRecord:
+            for dirKey, dirList in mongoRecord['dirs'].items():
+                mongoRecord['dirs'][dirKey] = set(dirList)
+            self.update(mongoRecord)
+            return True
+        else:
+            return False
+
+    def writeRSEToMongoDB(self, collection, dumpRSE=False, overwrite=False):
+        """
+        A method to write/update the RSE at the Database from the current object.
+        :param collection: The MonogoDB collection to write on
```

Comment:
BTW  `self.purgeRSEAtMongoDB` needs to be created, otherwise we will keep accumulating old records upon refresh of the RSE record from RucioConMon. I just forgot to put it in this commit. I was about to create and call the method in the PR related to `stateful` vs `stateless` service migration.

Reply:
Okay, we better leave this one (and a couple others) unresolved such that we can reference to this in the upcoming developments.